website/content/docs/v0.9/Guides/editing-machine-configuration.md

```diff
@@ -0,0 +1,101 @@
+---
+title: "Editing Machine Configuration"
+description: "How to edit and patch Talos machine configuration, with reboot, immediately, or stage update on reboot."
+---
+
+Talos node state is fully defined by [machine configuration](../Reference/configuration/).
+Initial configuration is delivered to the node at bootstrap time, but configuration can be updated while the node is running.
+
+> Note: Be sure that config is persisted so that configuration updates are not overwritten on reboots.
+> Configuration persistence was enabled by default since Talos 0.5 (`persist: true` in machine configuration).
+
+There are three `talosctl` commands which facilitate machine configuration updates:
+
+* `talosctl apply-config` to apply configuration from the file
+* `talosctl edit machineconfig` to launch an editor with existing node configuration, make changes and apply configuration back
+* `talosctl patch machineconfig` to apply automated machine configuration via JSON patch
+
+Each of these commands can operate in one of three modes:
+
+* apply change with a reboot (default): update configuration, reboot Talos node to apply configuration change
+* apply change immediately (`--immediate` flag): change is applied immediately without a reboot, only `.cluster` sub-tree of the machine configuration can be updated in Talos 0.9
+* apply change on next reboot (`--on-reboot`): change is staged to be applied after a reboot, but node is not rebooted
+
+> Note: applying change on next reboot (`--on-reboot`) doesn't modify current node configuration, so next call to
+> `talosctl edit machineconfig --on-reboot` will not see changes
+
+### `talosctl apply-config`
+
+This command is mostly used to submit initial machine configuration to the node (generated by `talosctl gen config`).
+It can be used to apply new configuration from the file to the running node as well, but most of the time it's not convenient, as it doesn't operate on the current node machine configuration.
+
+Example:
+
+```bash
+talosctl -n <IP> apply-config -f config.yaml
+```
+
+Command `apply-config` can also be invoked as `apply machineconfig`:
+
+```bash
+talosctl -n <IP> apply machineconfig -f config.yaml
+```
+
+Applying machine configuration immediately (without a reboot):
+
+```bash
+talosctl -n IP apply machineconfig -f config.yaml --immediate
+```
+
+### `taloctl edit machineconfig`
+
+Command `talosctl edit` loads current machine configuration from the node and launches configured editor to modify the config.
+If config hasn't been changed in the editor, configuration update is not applied.
```

Comment:
+ empty file also aborts the change.

Reply:
how can a user easily hit that? remove all the content? I could add this, just curious how can user hit that easily?

Reply:
just remove all contents, yeah :) It's the same for `kubectl edit`